libafl/src/state/mod.rs

```diff
@@ -532,6 +536,32 @@ where
     }
 }
 
+/// A trait for types that want to expose a stop API
+pub trait Stoppable {
+    /// Check if stop is requested
+    fn stop_requested(&self) -> bool;
+
+    /// Request to stop
+    fn request_stop(&mut self);
+
+    /// Discard the stop request
+    fn discard_stop_request(&mut self);
+}
+
+impl<I, C, R, SC> Stoppable for StdState<I, C, R, SC> {
+    fn request_stop(&mut self) {
```

Comment:
just a minor nit
shouldn't this be `stop_requested`?

because it's usually somebody else that request this fuzzer to stop right?

Reply:
sorry the name is already taken, then `set_stop_requested`?

Reply:
`stop_requested` is already used to see if we need to stop which makes more sense.
Also, we can request to stop ourselves in a stage or so. For eg for (``AFL_BENCH_UNTIL_CRASH``)

Reply:
For me the current names sound good